File: roboo/projects/move.cpp
#include <Arduino.h>
#include <Wire.h>
#include <LiquidCrystal_I2C.h>

LiquidCrystal_I2C lcd(0x27, 16, 2);


const char* lyrics[] = {
    "how can i",
    "move on?",
    "when",
    "i'm still",
    "in love <3",
    "with you!",
    "'cause if one day",
    "you wake up",
    "and find that you're",
    "missing me",
    "then your heart",
    "starts to wonder",
    "where on this earth",
    "i could be",
    "thinkin' maybe",
    "you'll come back",
    "here to the place",
    "that we'd meet",
    "and you'll see me",
    "waiting for you",
    "on the corner",
    "of the street",
    "so i'm not",
    "moving..",
    "i'm not moving"
};

// Array waktu tampil masing-masing lirik (ms)
const unsigned long durations[] = {
  1000,
  1000,
  1100,
  550,
  650,
  550,
  1500,
  520,
  800,
  1000,
  900,
  900,
  900,
  2000,
  800,
  1000,
  2000,
  1000,
  1200,
  1000,
  1000,
  1000,
  1000,
  1000,
  2500,
  5000
};
const int numLyrics = sizeof(lyrics)/sizeof(lyrics[0]);

void setup() {
  lcd.begin();
  lcd.backlight();
  lcd.clear();
  Serial.begin(9600);
  lcd.print("Waiting you...");
}

void loop() {
  if (Serial.available() > 0) {
    String cmd = Serial.readStringUntil('\n');
    cmd.trim();

    if (cmd == "move") {
      // jalankan lagu sekali
      for (int i = 0; i < numLyrics; i++) {
        lcd.clear();
        if (strlen(lyrics[i]) <= 16) {
          lcd.setCursor(0, 0);
          lcd.print(lyrics[i]);
          delay(durations[i]);
        } else {
          for (size_t j = 0; j <= strlen(lyrics[i])-16; j++) {
            lcd.setCursor(0,0);
            lcd.print(String(lyrics[i]).substring(j,j+16));
            delay(300);
          }
          delay(durations[i]);
        }
      }
      lcd.clear();
      lcd.print("Done");
    }
  }
}

File: script/move.py
import sys
import time

def moveon():
    lirik = [
        ("how can i move on", 0.1),
        ("when i'm still in love with you", 0.05),
        ("'cause if one day you wake up", 0.07),
        ("and find that you're missing me", 0.07),
        ("and your heart starts to wonder", 0.07),
        ("where on this earth i could be" ,0.06),
        ("thinkin' maybe you'll comeback", 0.08),
        ("here to the place that we'd meet", 0.06),
        ("and you'll see me waiting for you", 0.06),
        ("on the corner of the street", 0.09),
        ("so i'm not moving..", 0.08),
        ("i'm not moving...", 0.08)
    ]
    
    delay = [0.4, 0.3, 0.4, 0.3, 0.4, 0.4, 0.3, 0.4, 0.4, 0.4, 0.9, 0.9]
    time.sleep(2)
    
    for i, (lagu, delayk) in enumerate(lirik):
        for karakter in lagu:
            print(karakter, end='')
            sys.stdout.flush()
            time.sleep(delayk)
        time.sleep(delay[i])
        print('')
        
moveon()